### ISSUE TITLE
Compilation problem

### ISSUE THREAD

**Issue by hjotha** [LukasAddon/SpaceLemon-N910H-Kernel#3]
Lukas,

I've tryed building from the published sources of 2.3.2 and also from cloning your current repo
but im getting this error:

hjotha@hjotha-vbox:~/kernel/SpaceLemon-N910H-Kernel$ make trelte_00_defconfig
**arch/arm/Kconfig:930: can't open file "arch/arm/mach-mvebu/Kconfig"**
/home/hjotha/kernel/SpaceLemon-N910H-Kernel/scripts/kconfig/Makefile:105: recipe for target 'trelte_00_defconfig' failed
make[1]: *** [trelte_00_defconfig] Error 1
Makefile:517: recipe for target 'trelte_00_defconfig' failed
make: *** [trelte_00_defconfig] Error 2

hjotha@hjotha-vbox:~/kernel/SpaceLemon-N910H-Kernel$ make clean
scripts/Makefile.clean:17: /home/hjotha/kernel/SpaceLemon-N910H-Kernel/arch/arm/vfp/Makefile: No such file or directory
**make[1]: *** No rule to make target '/home/hjotha/kernel/SpaceLemon-N910H-Kernel/arch/arm/vfp/Makefile'.  Stop.**
Makefile:1075: recipe for target '_clean_arch/arm/vfp' failed
make: *** [_clean_arch/arm/vfp] Error 2

Are these files missing on git?
How to specify to build only the arch/arm/mach-exynos?

Best regards,
Henrique

**Comment by LukasAddon** [comment]
All files in git. 
in file /home/hjotha/kernel/SpaceLemon-N910H-Kernel/Makefile  in line 197  
->  CROSS_COMPILE	?=../arm-eabi-5.2/bin/arm-eabi-
did you change CROSS_COMPILE path	tool ?

**Comment by LukasAddon** [comment]
I see that git block some file in gitignore.
i push all lost files in next commit. it is  gitignore files in samsung sources.

**Comment by hjotha** [comment]
Your kernel is the best one there is. 
Ok, thank you very much!

**Comment by LukasAddon** [comment]
Done.
First in file /home/hjotha/kernel/SpaceLemon-N910H-Kernel/Makefile in line 197
-> CROSS_COMPILE	?=../arm-eabi-5.2/bin/arm-eabi-
write path to Linaro 5.2 

Then
Use command    ./build_kernel.sh to compile kernel

And you will see kernel in /build_kernel/out/SpaceLemon_v.zip

**Comment by LukasAddon** [comment]
if you have errors, just write me. i try to help you.

**Comment by LukasAddon** [comment]
Latest release v2.4.18  now on git

**Comment by hjotha** [comment]
Ok, I'll try. I'm using the same path to 5.2 tool chain as you (../arm-eabi-5.2/).So I don't need to edit the makefile. 
This build_kernel.sh builds the two versions of the kernel? I need to export something to build the standard version?  Or the extended battery version? 

Thanks

**Comment by LukasAddon** [comment]
no.  it build zerolemon version

but you can build standart version if rename 2 files.

for standart battery
/drivers/battery/max77843_fuelgauge.c   -  rename to /drivers/battery/max77843_fuelgauge_ZL.c
/drivers/battery/max77843_fuelgauge_ST.c   rename to /drivers/battery/max77843_fuelgauge.c

for zerolemon battery 
/drivers/battery/max77843_fuelgauge.c   -  rename to /drivers/battery/max77843_fuelgauge_ST.c
/drivers/battery/max77843_fuelgauge_ZL.c   rename to /drivers/battery/max77843_fuelgauge.c

i hope you understand me. 
i make build script later for this

**Comment by LukasAddon** [comment]
if you see
/drivers/battery/max77843_fuelgauge_ST.c = standart driver buttery
then 
/drivers/battery/max77843_fuelgauge.c  = zerolemon driver battery

**Comment by LukasAddon** [comment]
if you see
/drivers/battery/max77843_fuelgauge_ZL.c = zerolemon driver buttery
then
/drivers/battery/max77843_fuelgauge.c = standart driver battery

**Comment by LukasAddon** [comment]
max77843_fuelgauge  53.9 KB   = Zerolemon File
max77843_fuelgauge  51.9 KB   = Standart battery File

you can rename and complile two version of kernel

**Comment by hjotha** [comment]
Lukas,

I've tried many toolchains but I cant get my kernel to boot up:

from here:
https://releases.linaro.org/components/toolchain/binaries/

There are 3 versions of the 5.2 toolchain:

5.2-2015.11
5.2-2015.11-1
5.2-2015.11-2

I've compiled the kernel with these:

gcc-linaro-5.2-2015.11-x86_64_arm-linux-gnueabihf.tar.xz
gcc-linaro-5.2-2015.11-2-x86_64_arm-linux-gnueabi.tar.xz

But it is not booting.

Do you know exactly which version of the toolchain should I use?

Att,
Henrique

**Comment by LukasAddon** [comment]
5.2-2015.11-1 

If you receive  error in compiling, check log in console. maybe you have some error, and need to install some packets in system

**Comment by LukasAddon** [comment]
https://www.dropbox.com/s/cqiylbvvyl4otqz/arm-eabi-5.2.zip?dl=0   
my toolchain

**Comment by hjotha** [comment]
Lukas,

I have unpacked your kernel and the one I compiled.
Comparing the ramdisk it seems to be missing some files.
Mine on the left (19) yours on the right (18)
![image](https://cloud.githubusercontent.com/assets/23729943/20681928/e36e61a4-b58c-11e6-86ec-4d7037f5f5e4.png)

Maybe there are some files missing/different structure on?

SpaceLemon-N910H-Kernel/build_kernel/AIK-Linux/ramdisk/

Maybe git is not commiting empty directories?
Is it possible ? 
the /data folder is also missing 

Can you check yours?

Best regards,
Henrique

**Comment by hjotha** [comment]
it seems that one boot.img is different from the other.

**Comment by LukasAddon** [comment]
Ok i check ramdisk files.

**Comment by hjotha** [comment]
I was able to build and install the kernel without issues for this I had to:

Unpack the boot.img from the 2.4.18 release
Copy the ramdisk folder to
/home/hjotha/kernel/SpaceLemon-N910H-Kernel/build_kernel/AIK-Linux
Create a folder named data in 
/home/hjotha/kernel/SpaceLemon-N910H-Kernel/build_kernel/out

And then rebuild the kernel.

It seems there are uncommited files in the ramdisk folder, mostly empty directories
which I pointed in the previous SS.

I'm adding support for a cheap smart dock into the kernel
http://i.ebayimg.com/images/g/R6AAAOSwpdpVcvpL/s-l500.jpg

If I have progress i'll show you the changes.

Best regards,
Henrique
